packages/agents/src/agents/Chameleon/TextChunker.ts

```diff
@@ -0,0 +1,45 @@
+import { cleanWhitespace } from "./cleanWhitespace";
+
+export class TextChunker {
+  static singleLine(content: string): string[] {
+    const lines = cleanWhitespace(content).split("\n");
+    return lines;
+  }
+
+  static multiLines(content: string, linesPerChunk: number): string[] {
+    const lines = cleanWhitespace(content).split("\n");
+    const chunks = [];
+    for (let i = 0; i < lines.length; i += linesPerChunk) {
+      chunks.push(lines.slice(i, i + linesPerChunk).join("\n"));
+    }
+    return chunks;
+  }
+
+  static characters(content: string, characterLimit: number): string[] {
+    const trimmedContent = cleanWhitespace(content);
+    const chunks = [];
+    let currentChunk = "";
+    for (const char of trimmedContent) {
```

Comment:
Instead of iterating through each character, it would be faster to iterate for `trimmedContent.length % characterLimit` and use `substr` to extract a chunk of text.

Reply:
Sure